packages/lib-utils/src/app/api-discovery/index.ts

```diff
@@ -67,11 +67,11 @@ type APIResourceData = {
 };
 
 const batchResourcesRequest = (
-  batch: string[],
+  batch: (string | APIResourceList)[],
   groupVersionMap: AnyObject,
 ): Promise<DiscoveryResources>[] => {
-  return batch.map<Promise<DiscoveryResources>>(async (p: string) => {
-    const resourceList = await commonFetchJSON<APIResourceList>(p);
+  return batch.map<Promise<DiscoveryResources>>(async (p: string | APIResourceList) => {
+    const resourceList = typeof p === 'string' ? await commonFetchJSON<APIResourceList>(p) : p;
```

Comment:
Ideally I'd like a cleaner method of determining if we should fetch or not, but for now this should be fine (rather then fetching if the parameter is a string). When I was looking at this code I was wondering if we could split the actual fetch out to another function and have the processing be in it's own function that we could invoke directly with the static resource list.

Just a thought and not blocking.

Reply:
Yeah, I wanted to split the logic as well, but it was too complex to work with. Let's roll with this approach and we can refactor in current sprint as it's a tech debt sprint.